test/FsAutoComplete.Tests.Lsp/FindReferencesTests.fs

```diff
@@ -50,3 +58,1103 @@ let tests state =
+    let actual =
+      usages
+      |> Seq.map (fun u ->
+        Tokenizer.tryFixupRange(symbolNameCore, u.Range, source, includeBackticks)
+        |> Option.ofValueOption
+        |> Flip.Expect.wantSome $"Should be able to fixup usage '%A{u}'"
+      )
+      |> Seq.map fcsRangeToLsp
+      |> Seq.toArray
+      |> Array.sortBy (fun r -> (r.Start.Line, r.Start.Character))
+
+    let expected = expected |> Array.ofList
+
+    // Expect.equal actual expected "Should be correct range"
+    if actual <> expected then
+      // mark ranges for visual diff instead of range diff
+      let markRanges (ranges: Range[]) =
+        let locs =
+          ranges
+          |> Array.map (fun r ->
+              {
+                Uri = ""
+                Range = r
+              }
+          )
+        let marked = markRanges source.String locs
+        // append range strings for additional range diff
+        let rangeStrs =
+          ranges
+          |> Seq.map (fun r -> r.DebuggerDisplay)
+          |> String.concat "\n"
+        marked
+        + "\n"
+        + "\n"
+        + rangeStrs
+
+      Expect.equal
+        (markRanges actual)
+        (markRanges expected)
+        "Should be correct ranges" 
+  }
+
+  //TODO: why does this fail? Succeeds when run outside of FSAC
+  //  For example in Script file with `#r "nuget: FSharp.Compiler.Service, 41.0.5"`
+  // Note: It doesn't use any FSAC stuff -- just FCS
+  ptestCase "FCS: Active Pattern Case in Declaration" <| fun _ ->
+    let test () =
+      let checker = FSharp.Compiler.CodeAnalysis.FSharpChecker.Create()
+
+      let text = """
+let (|Even|Odd|) v = 
+  if v % 2 = 0 then Even else Odd
+match 2 with
+| Even -> ()
+| Odd -> ()
+    """
+      let file = "/code.fsx"
+      let source = FSharp.Compiler.Text.SourceText.ofString (text)
+      // decl
+      let pos = FSharp.Compiler.Text.Position.mkPos 3 24
+      // usage
+      // let pos = FSharp.Compiler.Text.Position.mkPos 5 6
+      let names = ["Even"]
+
+      let projOptions, _ =
+        checker.GetProjectOptionsFromScript (file, source)
+        |> Async.RunSynchronously
+      let (parseResults, checkResults') =
+        checker.ParseAndCheckFileInProject(file, 0, source, projOptions)
+        |> Async.RunSynchronously
+      let checkResults =
+        match checkResults' with
+        | FSharp.Compiler.CodeAnalysis.FSharpCheckFileAnswer.Succeeded checkResults -> checkResults
+        | _ -> failwith "CheckFile -> aborted"
+
+      let symbolUse = 
+        checkResults.GetSymbolUseAtLocation(pos.Line, pos.Column, source.GetLineString (pos.Line-1), names)
+        |> Option.defaultWith (fun _ -> failwith "no symbol at location")
+
+      printfn "DisplayNameCore=%A" symbolUse.Symbol.DisplayNameCore
+      printfn "FullName=%A" symbolUse.Symbol.FullName
+
+      if symbolUse.Symbol.DisplayNameCore = "" then
+        failwith $"No display name for symbol with full name '{symbolUse.Symbol.FullName}'"
+
+    test()
```

Comment:
There's a strange case for Active Pattern Case and `GetSymbolUseAtLocation`:
```fsharp
let (|Even|Odd|) v =
  if v % 2 = 0 then Even else Odd
  //                ^^^^ decl
match 42 with
| Even -> ()
//^^^^ usage
| Odd -> ()
```

Getting Symbol for `Even` at usage (in match pattern) returns correct Symbol with correct `DisplayNameCore` (`Even`).  
But getting Symbol inside Active Pattern Declaration returns Symbol with empty `DisplayNameCore` -- but `FullName` is correct: `FullName = Even`. So it points to correct Symbol -- just `DisplayName` not correctly populated.

So seems like FCS bug. Reproduced here in this test  
(to run: remove `p` (`ptestCase` -> `testCase`) and run `dotnet run --project ./test/FsAutoComplete.Tests.Lsp --framework net6.0 -- --run 'FSAC.general.tryFixupRange.FCS: Active Pattern Case in Declaration'`)  
Note: the test doesn't use any FSAC stuff but just FSharp Compiler Service

Now the really strange thing: this only happens in FSAC.  
Running the same code outside of FSAC does work (`DisplayNameCore = Even`) 
<details>
  <summary>To run in Script File</summary>
  
  ```fsharp
  // Note: same version FSAC currently uses
  #r "nuget: FSharp.Compiler.Service, 41.0.5"

  let test () =
        let checker = FSharp.Compiler.CodeAnalysis.FSharpChecker.Create()

        let text = """
  let (|Even|Odd|) v = 
    if v % 2 = 0 then Even else Odd
  match 2 with
  | Even -> ()
  | Odd -> ()
      """
        let file = "/code.fsx"
        let source = FSharp.Compiler.Text.SourceText.ofString (text)
        // decl
        let pos = FSharp.Compiler.Text.Position.mkPos 3 24
        // usage
        // let pos = FSharp.Compiler.Text.Position.mkPos 5 6
        let names = ["Even"]


        let projOptions, _ =
          checker.GetProjectOptionsFromScript (file, source)
          |> Async.RunSynchronously
        let (parseResults, checkResults') =
          checker.ParseAndCheckFileInProject(file, 0, source, projOptions)
          |> Async.RunSynchronously
        let checkResults =
          match checkResults' with
          | FSharp.Compiler.CodeAnalysis.FSharpCheckFileAnswer.Succeeded checkResults -> checkResults
          | _ -> failwith "CheckFile -> aborted"

        let symbolUse = 
          checkResults.GetSymbolUseAtLocation(pos.Line, pos.Column, source.GetLineString (pos.Line-1), names)
          |> Option.defaultWith (fun _ -> failwith "no symbol at location")

        printfn "DisplayNameCore=%A" symbolUse.Symbol.DisplayNameCore
        printfn "FullName=%A" symbolUse.Symbol.FullName

        if symbolUse.Symbol.DisplayNameCore = "" then
          failwith $"No display name for symbol with full name '{symbolUse.Symbol.FullName}'"

  test()
  ```
</details>


So it doesn't seem to be an issue with FCS -- but with FSAC? I have no idea

Reply:
Yeah probably worth reporting to `dotnet/fsharp`. (@baronfel might also have some insight as to if this might be different for the latest FCS version)

Reply:
using FCS 43.7.200 (the one we'll use in the .NET 7 PR), the output is

```
DisplayNameCore="Even"
FullName="Even"
```

So it looks like it's fixed?

Reply:
Yes: Can confirm it works with #1043